Yubico.NativeShims/build-macOS.sh

```diff
@@ -7,13 +7,13 @@ git pull
 vcpkg x-update-baseline
 popd
 
-\cmake \
+cmake \
     -S . \
     -B ./build64 \
     -DCMAKE_BUILD_TYPE=Release \
     -DCMAKE_TOOLCHAIN_FILE=$VCPKG_INSTALLATION_ROOT/scripts/buildsystems/vcpkg.cmake \
-    -DVCPKG_TARGET_TRIPLET=x64-osx
-
+    -DVCPKG_TARGET_TRIPLET=x64-osx \
+    -DCMAKE_OSX_ARCHITECTURES=x86_64
```

Comment:
At one point we used custom toolchain files. It had set the `CMAKE_OSX_ARCHITECTURE` option among others. We probably unintentionally stopped using them when we started using VCPKG.

I'd recommend taking a look at the ones we have in the `cmake/` subfolder. We should probably either start using them as overlays (I think that's the term vcpkg uses) or get rid of them.

Either way - would probably be good to check to see if there are other settings that were used once upon a time that might be worth reintroducing.

I'd also double check to see if `VCPKG_TARGET_TRIPLET` doesn't already set/imply `CMAKE_OSX_ARCHITECTURES` if you haven't done so already.

Reply:
Good points! I did some investigations here and this is what I found out:

Although the triplets set `VCPKG_OSX_ARCHITECTURES` to correct values, these is not used at all on OSX leaving `CMAKE_OSX_ARCHITECTURES` undefined, which will fail linkage for specific build/host/target combinations. Explicitly defining the value for each target platform fixes the issue.

I was not aware of the `cmake/osx*cmake` files, currently they are not used and in the original commit they served as the main toolchain files. They could be reused through `VCPKG_CHAINLOAD_TOOLCHAIN_FILE` but I don’t see any advantage to just using `-DCMAKE_OSX_ARCHITECTURES` as vcpkg is used now.

See also that the `cmake/osx*cmake` files do cache the specific `CMAKE_OSX_ARCHITECTURES` value (functionally same effect as using the explicit `-D`).

My way forward would be to use `-DCMAKE_OSX_ARCHITECTURES=arch` and delete the unused cmake files. What do you think?

Reply:
Yes, that sounds quite reasonable to me. Thanks Adam!

Reply:
+1 for simplifying and deleting unused files. Thanks @AdamVe !